### PR TITLE
Host resource ping method should return stdout

### PR DESCRIPTION
**Pull request by justincmoy**
Prior to this PR, the ping method would return the following for nc (stderr output):
```
> echo | nc -v -w 1 127.0.0.1 1900 1>/dev/null
Ncat: Version 6.40 ( http://nmap.org/ncat )
Ncat: Connected to 127.0.0.1:1900.
Ncat: Connection reset by peer.
```

After this PR, the ping method would return the following for nc (stdout output):
```
> echo | nc -v -w 1 127.0.0.1 1900 2>/dev/null
DOWN
```

### CODE REVIEW

**Comment by adamleff** [comment]
@justincmoy I'm 👎  on this as-written because the STDERR from netcat is actually quite useful in troubleshooting, and the STDOUT isn't nearly as useful, especially since many far-end targets will not respond with any data when you send an empty message.

Can you explain the reason for this proposed change?

**Comment by justincmoy** [comment]
@adamleff Sure thing. We have an haproxy agent-check that connects to an xinetd daemon, similar to this blog post: https://icicimov.github.io/blog/server/HAProxy-dynamically-adjust-server-weight-using-external-agent/

We would like to write an inspec test that verifies connecting to our specified xinetd port returns expected output.

Edit:
Would it make sense to have these methods return an additional key corresponding to the stdout? And then a public method that can access it?

**Comment by adamleff** [comment]
Thanks for sharing your use case. It makes the case for exposing both sets of content.

I'd like to avoid using `stdout` and `stderr` as the method names since we may change the implementation in the future and an end-user shouldn't need to know how we gathered the data. And if we want to provide both sets of output to the user, `output` isn't specific enough anymore.

What if we call the `stderr` output `connection_output` and the `stdout` output `socket_output` and have method names that match that?

**Comment by justincmoy** [comment]
👍  I'll update this PR accordingly

**Review by adamleff — APPROVED** [review]
@justincmoy this looks great, thank you. :)

File: lib/resources/host.rb [hunk 1]
```diff
@@ -156,7 +160,8 @@ def ping(hostname, port, protocol)
 
       {
         success: resp.exit_status.to_i.zero?,
-        output: resp.stderr,
+        connection_output: resp.stderr,
```
Comment (chris-rock) [inline lib/resources/host.rb]: why not call it straight `stdout` and `stderr`

Comment (justincmoy) [reply lib/resources/host.rb]: This came up in an earlier comment. It's to hide the implementation details from the end user.

Comment (chris-rock) [reply lib/resources/host.rb]: Got ya. Thank you, I misread this

Comment (adamleff) [reply lib/resources/host.rb]: As stated in https://github.com/chef/inspec/pull/1927#issuecomment-308794898.  Netcat's `stdout` and `stderr` is what we're using now, but if we iterate on this in the future and find a tool that doesn't use `stdout`/`stderr` the same way, then we're training the user on a specific implementation.

There are two bits of data being exposed: output about the connection setup, and the output from the socket itself after the connection is established.

I proposed `connection_output` and `socket_output` accordingly, and I think the `_output` piece is important so someone doesn't believe these can be manipulated -- it's simply the output of the connection attempt. I'm totally fine with removing `_output` if it helps calm any concerns.

**Comment by adamleff** [comment]
@justincmoy after an internal discussion, would you mind changing the method names and the hash keys to remove the `_output` from them?

`connection_output` => `connection`
`socket_output` => `socket`

**Review by adamleff — APPROVED** [review]

**Review by chris-rock — APPROVED** [review]
Thank you @justincmoy and @adamleff

**Pull request merged**